tests/datasource/fluent/integration/test_integration_datasource.py

```diff
@@ -483,15 +487,20 @@ def test_pandas_data_adding_dataframe_in_file_reloaded_context(
 
     datasource = context.data_sources.add_or_update_pandas(name="fluent_pandas_datasource")
     dataframe_asset: PandasDataFrameAsset = datasource.add_dataframe_asset(name="my_df_asset")
-    _ = dataframe_asset.build_batch_request(dataframe=df)
-    assert dataframe_asset.dataframe.equals(df)  # type: ignore[attr-defined] # _PandasDataFrameT
+    batch_def = dataframe_asset.add_batch_definition_whole_dataframe(name="bd")
+    batch = batch_def.get_batch(batch_parameters={"dataframe": df})
+    assert isinstance(batch.data, PandasBatchData)
+    assert batch.data.dataframe.equals(df)
 
+    # Reload the asset and see that we can re-add the df to the batch definition
     context = gx.get_context(context_root_dir=context.root_directory, cloud_mode=False)
     dataframe_asset = context.get_datasource(datasource_name="fluent_pandas_datasource").get_asset(
         asset_name="my_df_asset"
     )
-    _ = dataframe_asset.build_batch_request(dataframe=df)
-    assert dataframe_asset.dataframe.equals(df)  # type: ignore[attr-defined] # _PandasDataFrameT
+    reloaded_batch_def = dataframe_asset.get_batch_definition(batch_definition_name="bd")
```

Comment:
We should change the argument to `name` instead of `batch_definition_name`. I can do that in a separate PR.

Reply:
I actually see both patterns in the code depending on domain object. I'd prefer these to all be name since we know the type of the object from the method name, eg `get_<domain_object_type>`.